packages/firebase_storage/firebase_storage/android/src/main/java/io/flutter/plugins/firebase/storage/FlutterFirebaseStoragePlugin.java

```diff
@@ -133,14 +135,61 @@ public void onAttachedToEngine(@NonNull FlutterPluginBinding binding) {
   public void onDetachedFromEngine(@NonNull FlutterPluginBinding binding) {
     FlutterFirebaseStorageTask.cancelInProgressTasks();
     channel.setMethodCallHandler(null);
+    assert messenger != null;
+    GeneratedAndroidFirebaseStorage.FirebaseStorageHostApi.setup(messenger, null);
     channel = null;
+    messenger = null;
+    removeEventListeners();
```

Comment:
`removeEventListeners()` also needs to be called within `didReinitializeFirebaseCore()` method.

Reply:
done